File: sdk/dart/test/put_flag_request_test.dart
import 'package:test/test.dart';
import 'package:flagent_client/flagent_client.dart';

// tests for PutFlagRequest
void main() {
  final instance = PutFlagRequest((b) => b
    ..description = 'Updated description'
    ..key = 'updated_key'
    ..dataRecordsEnabled = true
    ..entityType = 'device'
    ..notes = 'Updated notes');

  group(PutFlagRequest, () {
    test('to test the property `description`', () async {
      expect(instance.description, equals('Updated description'));
    });

    test('to test the property `key`', () async {
      expect(instance.key, equals('updated_key'));
    });

    test('to test the property `dataRecordsEnabled`', () async {
      expect(instance.dataRecordsEnabled, equals(true));
    });

    test('to test the property `entityType`', () async {
      expect(instance.entityType, equals('device'));
    });

    test('to test the property `notes`', () async {
      expect(instance.notes, equals('Updated notes'));
    });
  });
}

File: sdk/java/src/test/java/com/flagent/client/model/PutFlagRequestTest.java
package com.flagent.client.model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Model tests for PutFlagRequest
 */
class PutFlagRequestTest {

    @Test
    void testPutFlagRequest() {
        PutFlagRequest model = new PutFlagRequest()
            .description("Updated")
            .key("key")
            .dataRecordsEnabled(true)
            .entityType("user")
            .notes("Notes");
        Assertions.assertNotNull(model);
        Assertions.assertEquals("Updated", model.getDescription());
        Assertions.assertEquals("key", model.getKey());
        Assertions.assertTrue(model.getDataRecordsEnabled());
        Assertions.assertEquals("user", model.getEntityType());
        Assertions.assertEquals("Notes", model.getNotes());
    }

    @Test
    void descriptionTest() {
        PutFlagRequest model = new PutFlagRequest().description("Desc");
        Assertions.assertEquals("Desc", model.getDescription());
    }

    @Test
    void keyTest() {
        PutFlagRequest model = new PutFlagRequest().key("my_key");
        Assertions.assertEquals("my_key", model.getKey());
    }

    @Test
    void dataRecordsEnabledTest() {
        PutFlagRequest model = new PutFlagRequest().dataRecordsEnabled(true);
        Assertions.assertTrue(model.getDataRecordsEnabled());
    }

    @Test
    void entityTypeTest() {
        PutFlagRequest model = new PutFlagRequest().entityType("device");
        Assertions.assertEquals("device", model.getEntityType());
    }

    @Test
    void notesTest() {
        PutFlagRequest model = new PutFlagRequest().notes("Note");
        Assertions.assertEquals("Note", model.getNotes());
    }

}
